### ISSUE TITLE
Smarter flatpakref handling

### ISSUE THREAD

**Issue by teamcons** [elementary/triage#675]
### Problem

Sometimes i wade through the flathub website to discover, instead of appcenter
clicking on "install" downloads a flatpakref
double-clicking the flatpakref opens sideload, with all the freaky warnings, despite flathub being added already

### Proposal

Detect on opening with a flatpakref whether:
-it is from an already installed repo -> Open the corresponding page in appcenter
-it isnt -> all the warnings from sideload

i am not sure how feasible it would be to have the whole thing by clicking on "install" instead of downloading the flatpakref first. Thats probably out of the eOS teams hands.

**Comment by ryonakano** [comment]
I agree with you that it's a little strange to warn apps from Flathub, but it wouldn't possible to switch an app to open with because an app to open with a file and that file is related with the file extension, i.e. .flatpakref. We would need to resolve this issue inside one app.

**Comment by teamcons** [comment]
My line of thought was - we could have appcenter be the solver
it checks the URL line against the repos it has. And summon sideloader or open corresponding page depending

It would make sense to have appcenter be "responsible" for opening flatpakref
sideload isnt really an app by itself. For a user, it is more like a system dialog.

Or there could be a simple app solver as default for flatpakref
